pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -309,6 +306,13 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 		}
 		return nil
 	}
+
+	if pipelineState.IsDone() && pr.IsDone() {
+		c.timeoutHandler.Release(pr)
+		c.Recorder.Event(pr, corev1.EventTypeNormal, eventReasonSucceeded, "PipelineRun completed successfully.")
+		return nil
+	}
```

Comment:
why did this block need to move to much later in the function?

Reply:
in order to have pipeline state fulfilled with task run data.